tee_supplicant/src/ree_fs.c

```diff
@@ -195,16 +195,24 @@ static int tee_fs_read(size_t num_params, struct tee_param *params)
 		return TEEC_ERROR_BAD_PARAMETERS;
 	}
 	len = MEMREF_SIZE(params + 1);
-	rc = fs_seek(file, offset, SEEK_SET);
-	if (rc < 0) {
-		LOG_ERR("invalid offset %ld (%d)", offset, rc);
-		return TEEC_ERROR_ITEM_NOT_FOUND;
-	}
-
-	sz = fs_read(file, buf, len);
-	if (sz < 0) {
-		LOG_ERR("read failure (%ld)", sz);
-		return TEEC_ERROR_GENERIC;
+	sz = 0;
+	while (len > 0) {
+		rc = fs_seek(file, offset+sz, SEEK_SET);
```

Comment:
Formatting: `offset + sz`

Reply:
Also, why are you doing seek on every iteration? IMO, you need to make it only once and then call `fs_read` inside the loop